schema/map_test.go

```diff
@@ -362,3 +363,49 @@ func TestMap_UnSerialize_Reversible(t *testing.T) {
 	// test reversiblity
 	assert.Equals(t, unserialized2, unserialized)
 }
+
+// Test type aliased map of type aliased string to a struct.
+type TypedStringEnumTestType string
+
+const (
+	testA TypedStringEnumTestType = "testA"
+	testB TypedStringEnumTestType = "testB"
+)
+
+func TestMap_AliasedTypes(t *testing.T) {
+	schemaForPrivateFieldStruct := schema.NewStructMappedObjectSchema[testdata.TestStructWithPrivateField](
+		"structWithPrivateField",
+		map[string]*schema.PropertySchema{
+			"field1": schema.NewPropertySchema(
+				schema.NewStringSchema(nil, nil, nil),
+				nil,
+				false,
+				nil,
+				nil,
+				nil,
+				schema.PointerTo("\"Hello world!\""),
+				nil,
+			),
+		},
+	)
+	schemaForEnum := schema.NewTypedStringEnumSchema[TypedStringEnumTestType](map[TypedStringEnumTestType]*schema.DisplayValue{
+		testA: nil,
+		testB: nil,
+	})
+	mapType := schema.NewMapSchema(
+		schemaForEnum,
+		schemaForPrivateFieldStruct,
+		nil,
+		nil,
+	)
+
+	unserializedData, err := mapType.Unserialize(map[string]any{
+		string(testA): map[string]any{
+			"field1": "test_field_value",
+		},
+	})
+	assert.NoError(t, err)
+	assert.InstanceOf[map[TypedStringEnumTestType]testdata.TestStructWithPrivateField](t, unserializedData)
+	_, err = mapType.Serialize(unserializedData)
```

Comment:
I realize that the point of this code is to handle the *type* rather than worrying about preservation of *values*, but it always bugs me to see test code that only cares about "operation success" without doing anything to verify that the "successful" operation was correct. That is, I'd love to see proof that `field1` is `"test_field_value"`